src/login/logind-dbus.c

```diff
@@ -1958,6 +1958,8 @@ static int method_do_shutdown_or_sleep(
 static int method_poweroff(sd_bus_message *message, void *userdata, sd_bus_error *error) {
         Manager *m = userdata;
 
+        m->scheduled_shutdown_type = strdup("poweroff");
```

Comment:
Maybe, add one more argument to `method_do_shutdown_or_sleep()` to take type string?

Also, it is better to use `free_and_strdup()`.

Reply:
I see. Then, `free_and_strdup()` sounds enough.